src/Data/Tree/Binary/Properties.agda

```diff
@@ -48,3 +51,16 @@ private
 map-id : ∀ (t : Tree N L) → map id id t ≡ t
 map-id (leaf x)     = refl
 map-id (node l v r) = cong₂ (flip node v) (map-id l) (map-id r)
+
+map-compose : ∀ {f₁ : N₁ → N₂} {f₂ : N → N₁} {g₁ : L₁ → L₂} {g₂ : L → L₁} →
+              map (f₁ ∘ f₂) (g₁ ∘ g₂) ≗ map f₁ g₁ ∘ map f₂ g₂
+map-compose (leaf x) = refl
+map-compose (node l v r) = cong₂ (λ l r → node l _ r) (map-compose l) (map-compose r)
+
+map-cong : ∀ {f₁ f₂ : N → N₁} {g₁ g₂ : L → L₁} → f₁ ≗ f₂ → g₁ ≗ g₂ → map f₁ g₁ ≗ map f₂ g₂
+map-cong p q (leaf x) = cong leaf (q x)
+map-cong p q (node l v r) = recombine (map-cong p q l) (p v) (map-cong p q r)
```

Comment:
`recombine` is [`congₙ`](https://agda.github.io/agda-stdlib/Function.Nary.NonDependent.html#2015) applied to `node`.

Reply:
Nice, I didn't know about this function!